src/assets/_project/_blocks/components/forms/recaptcha.js

```diff
@@ -81,10 +91,44 @@ import keys from '../../data/qg-google-keys';
   };
 
   let loadedRecaptcha = false;
+  let onloadRecaptchaAjax = () => {
+    grecaptcha.ready(function () {
+      $('#qg-page-feedback-form').submit(function (event) {
+        var self = $(this);
+        event.preventDefault();
+        if ($('#qg-page-feedback-form li.invalid').length <= 0) {
+          var greptcha = $feedbackForm.find('input[name="g-recaptcha-response"]');
+          var postUrl = self.attr('action');
+          var requestMethod = self.attr('method');
+          var $successMsgContainer = $('.thankyou');
+          grecaptcha.execute(footerFeedbackGoogleRecaptchaApiKey, {action: 'feedback'})
+            .then(function (token) {
+              if (greptcha.length > 0) {
+                greptcha.val(token);
+                var formData = self.serialize();
+                $.ajax({
+                  url: postUrl,
+                  type: requestMethod,
+                  data: formData,
+                  contentType: 'application/x-www-form-urlencoded',
+                  cache: false,
+                  processData: false,
+                }).done(function (response) {
```

Comment:
what is displayed if it errors?

Reply:
How submission handles the errors because I think it just show Recaptcha error not actually the errors?

I can place code in jquery ajax like this but I think this works only if server handles the error and send appropriate info(status) etc.

`.fail(function (xhr, status, error) {
                  var errorMessage = xhr.status + ': ' + xhr.statusText;
                  console.log('Error - ' + errorMessage);
                  $successMsgContainer.append('<p>Request failed, please try again</p>');
   });`

Reply:
might not be good putting the full payload in console as its a full html page when it errors. unless you grep for recaptcha or the block.

Reply:
I am just displaying the message $successMsgContainer.append('<p>Request failed, please try again</p>'); on error 

removed console log messages

Reply:
if you don't get a json payload and it error, you could look for the qg-primary-content element in the returned page and put that text out. or just the sublines

document.querySelector("#qg-primary-content > h1")
document.querySelector("#qg-primary-content > div.alert.alert-warning > div > h2")
document.querySelector("#qg-primary-content > div.alert.alert-warning > div > ul > li > span")

h1 -> An error has occurred
h2 -> Unable to continue
span -> reCAPTCHA Missing, the action will now terminate and no email will be sent.

https://test.smartservice.qld.gov.au/services/submissions/email/feedback/feedback-v2

Reply:
Developers can look in the network console to find any errors for debugging purposes
Why we put in HTML?  I think we just need a friendly message for customers to 'try again' ?

Reply:
Hi William, 
I don't think of any straightforward method to distinguish error. Also in my testing whenever I submit a form with mandatory field missing etc I always get Recaptcha error only on one occasion when submission handler was down I got a different error that server is down.